src/hotspot/share/oops/cpCache.cpp

```diff
@@ -716,11 +717,9 @@ void ConstantPoolCache::deallocate_contents(ClassLoaderData* data) {
   set_resolved_references(OopHandle());
   MetadataFactory::free_array<u2>(data, _reference_map);
   set_reference_map(NULL);
-
 #if INCLUDE_CDS
-  if (Arguments::is_dumping_archive()) {
-    SystemDictionaryShared::remove_saved_cpcache_entries(this);
-  }
+  MetadataFactory::free_array<ConstantPoolCacheEntry>(data, _initial_entries);
+  _initial_entries = NULL;
```

Comment:
Should the above be inside the following condition like before?

`if (Arguments::is_dumping_archive()) {`

Reply:
I added a more strict check -- since _initial_entries is set only during CDS dump time, I added an assert that if _initial_entries  is non NULL, we must be dumping.

Reply:
The non NULL check is good.